contentgrid-spring-boot-autoconfigure/src/main/java/com/contentgrid/spring/boot/autoconfigure/security/MultiTenantOAuth2ResourceServerAutoConfiguration.java

```diff
@@ -0,0 +1,44 @@
+package com.contentgrid.spring.boot.autoconfigure.security;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication.Type;
+import org.springframework.boot.autoconfigure.security.oauth2.resource.servlet.OAuth2ResourceServerAutoConfiguration;
+import org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.authentication.AuthenticationManagerResolver;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.oauth2.server.resource.authentication.JwtIssuerAuthenticationManagerResolver;
+import org.springframework.security.web.SecurityFilterChain;
+
+@AutoConfiguration(before = { OAuth2ResourceServerAutoConfiguration.class, SecurityAutoConfiguration.class })
+@EnableConfigurationProperties(MultiTenantOAuth2Properties.class)
+@ConditionalOnClass(value = {JwtIssuerAuthenticationManagerResolver.class, SecurityFilterChain.class, HttpSecurity.class})
+@ConditionalOnWebApplication(type = Type.SERVLET)
+class MultiTenantOAuth2ResourceServerAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnProperty("contentgrid.security.oauth2.trusted-jwt-issuers[0]")
+    JwtIssuerAuthenticationManagerResolver authenticationManagerResolver(MultiTenantOAuth2Properties oauth2Properties) {
+        return new JwtIssuerAuthenticationManagerResolver(oauth2Properties.getTrustedJwtIssuers());
+    }
+
+    @Bean
+    @ConditionalOnBean(JwtIssuerAuthenticationManagerResolver.class)
+    SecurityFilterChain jwtIssuerSecurityFilterChain(HttpSecurity http,
+            AuthenticationManagerResolver<HttpServletRequest> authManagerResolver) throws Exception {
```

Comment:
Conditional on a bean of a certain type being present, but the object that's injected is a generic interface.

That means that it is possible for a different `AuthenticationManagerResolver` to be injected here, which would be unexpected. (multiple beans of the same type won't result in an error if one of them is marked `@Primary`)

Reply:
fair, I'll specialise the argument so they match